pika/adapters/select_connection.py

```diff
@@ -264,21 +264,38 @@ def get_next_deadline(self):
         timeout = min((timeout, SelectPoller.POLL_TIMEOUT))
         return timeout * SelectPoller.POLL_TIMEOUT_MULT
 
+    @staticmethod
+    def is_ready_to_fire(to_run):
+        """For patching by unittest"""
+        return True
+
     def process_timeouts(self):
         """Process the self._timeouts event stack"""
 
         now = time.time()
-        to_run = [timer for timer in self._timeouts.values()
-                  if timer['deadline'] <= now]
-
         # Run the timeouts in order of deadlines. Although this shouldn't
         # be strictly necessary it preserves old behaviour when timeouts
         # were only run periodically.
-        for t in sorted(to_run, key=itemgetter('deadline')):
```

Comment:
How should I deal with the warnings like one below?
  Attribute 'timer_stack' defined outside __init__ (attribute-defined-outside-init)
If we initialize all attributes, that are need only temporarily during execution of certain test(s), then __init__ method gets too messy. Any suggestions?

Another thing is I don't see pylint complain about the use of assertEqual(). Will you please show me how you run pylint? Or the comment "Please check your changes with pylint and fix findings." isn't necessarily directed to the diff that the comment is attached to?